urlchecker/core/check.py

```diff
@@ -51,19 +53,33 @@ def __init__(
         self.file_types = file_types or [".py", ".md"]
         self.file_paths = []
 
-        # get all file paths if a path is defined
-        if path:
-
-            # Exit early if path not defined
-            if not os.path.exists(path):
-                sys.exit("%s does not exist." % path)
-
-            self.file_paths = fileproc.get_file_paths(
-                include_patterns=self.include_patterns,
-                base_path=path,
-                file_types=self.file_types,
-                exclude_files=self.exclude_files,
-            )
+        if files:
+            # Check that all files specified exist and are not directories
+            for file in files:
+                # Exit early if path not defined
+                if not os.path.exists(file):
```

Comment:
os.path.isfile should check for both existence and that it's not a directory - you don't need to double check.